extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/ChangeRecordImpl.java

```diff
@@ -22,92 +22,78 @@
 import com.hazelcast.jet.cdc.RecordPart;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.Serializable;
 import java.util.Map;
-import java.util.Objects;
 
-public class ChangeRecordImpl implements ChangeRecord {
+import static java.util.Objects.requireNonNull;
+
+public class ChangeRecordImpl implements ChangeRecord, Serializable {
+    private static final long serialVersionUID = 1L;
 
     private final long sequenceSource;
     private final long sequenceValue;
     private final String keyJson;
-    private final String valueJson;
-
-    private String json;
-    private Long timestamp;
-    private Operation operation;
-    private String database;
-    private String schema;
-    private String table;
+
+    private final Long timestamp;
+    private final Operation operation;
+    private final String database;
+    private final String schema;
+    private final String table;
     private RecordPart key;
-    private RecordPart value;
+    private final RecordPart oldValue;
+    private final RecordPart newValue;
 
     public ChangeRecordImpl(
+            long timestamp,
             long sequenceSource,
             long sequenceValue,
+            Operation operation,
             @Nonnull String keyJson,
-            @Nonnull String valueJson
+            @Nullable String oldValueJson,
+            @Nullable String newValueJson,
+            String table,
+            String schema,
+            String database
     ) {
+        this.timestamp = timestamp;
         this.sequenceSource = sequenceSource;
         this.sequenceValue = sequenceValue;
-        this.keyJson = Objects.requireNonNull(keyJson, "keyJson");
-        this.valueJson = Objects.requireNonNull(valueJson, "valueJson");
+        this.operation = operation;
+        this.keyJson = requireNonNull(keyJson, "keyJson");
+        this.oldValue = oldValueJson == null ? null : new RecordPartImpl(oldValueJson);
+        this.newValue = newValueJson == null ? null : new RecordPartImpl(newValueJson);
```

Comment:
Could these be lazy?

Reply:
I think so, but the RecordPartImpl constructor doesn't really do anything. just saves JSON value.  all the conversions are lazy, so is it worth making it lazy here too?